### ISSUE TITLE
Empty `generated_text` when tokens end in middle of arabic unicode character 

### ISSUE THREAD

**Issue by Vinno97** [huggingface/text-generation-inference#1235]
### System Info

Running in docker, using the (as of the moment of writing) the most recent version of `ghcr.io/huggingface/text-generation-inference:latest`.
```
root@f1772177cec6:/usr/src# text-generation-launcher --env
2023-11-05T19:24:58.321533Z  INFO text_generation_launcher: Runtime environment:
Target: x86_64-unknown-linux-gnu
Cargo version: 1.70.0
Commit sha: 96a982ad8fc232479384476b1596a880697cc1d0
Docker label: sha-96a982a
nvidia-smi:
Sun Nov  5 19:24:58 2023       
   +---------------------------------------------------------------------------------------+
   | NVIDIA-SMI 535.129.03             Driver Version: 535.129.03   CUDA Version: 12.2     |
   |-----------------------------------------+----------------------+----------------------+
   | GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
   | Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
   |                                         |                      |               MIG M. |
   |=========================================+======================+======================|
   |   0  NVIDIA GeForce RTX 3060        Off | 00000000:08:00.0  On |                  N/A |
   |  0%   50C    P5              22W / 170W |  11216MiB / 12288MiB |     55%      Default |
   |                                         |                      |                  N/A |
   +-----------------------------------------+----------------------+----------------------+
                                                                                            
   +---------------------------------------------------------------------------------------+
   | Processes:                                                                            |
   |  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
   |        ID   ID                                                             Usage      |
   |=======================================================================================|
   +---------------------------------------------------------------------------------------+
```

```json
{
  "model_id": "bigcode/santacoder",
  "model_sha": "bb3be599767d93ce716293e9193c027e855a9524",
  "model_dtype": "torch.float16",
  "model_device_type": "cuda",
  "model_pipeline_tag": "text-generation",
  "max_concurrent_requests": 128,
  "max_best_of": 2,
  "max_stop_sequences": 4,
  "max_input_length": 1024,
  "max_total_tokens": 2048,
  "waiting_served_ratio": 1.2,
  "max_batch_total_tokens": 561120,
  "max_waiting_tokens": 20,
  "validation_workers": 2,
  "version": "1.1.1",
  "sha": "96a982ad8fc232479384476b1596a880697cc1d0",
  "docker_label": "sha-96a982a"
}
```

### Information

- [X] Docker
- [ ] The CLI directly

### Tasks

- [X] An officially supported command
- [ ] My own modifications

### Reproduction

I'm trying to use TGI as a back-end for lm-eval, where I found out that TGI has a bug in its (token) string concatenation for some of the benchmarks. I'm pretty sure it has something to do with multi-byte Unicode characters and I can easily trigger it using the following request (to the aforementioned starcoder model, though I don't assume it's model-dependent):

1. Start TGI with these settings:
   
   ```yaml
   version: "3.3"
   
   services:
     tgi:
       image: ghcr.io/huggingface/text-generation-inference:latest
       ports:
         - 8080:80
       volumes:
         - ./tgi-mount:/data
       command:
         --model-id bigcode/santacoder
       deploy:
         resources:
           reservations:
             devices:
               - driver: nvidia
                 count: 1
                 capabilities: [gpu]
   ```

2. Send the following request
   ```sh
   curl 127.0.0.1:8080/generate \
       -X POST \
       -d '{"inputs": "English phrase: Guess what?\nArabic phrase:","parameters":{"max_new_tokens":256, "details": true}}' \
       -H 'Content-Type: application/json'
   ```

3. Observer `"generated_text": ""` in the output
   <details>

   ```json
   {
   "generated_text": "",
   "details": {
      "finish_reason": "length",
      "generated_tokens": 256,
      "seed": null,
      "prefill": [],
      "tokens": [
         { "id": 207, "text": " ", "logprob": -1.0556641, "special": false },
         { "id": 14709, "text": "ت", "logprob": -2.1035156, "special": false },
         { "id": 146, "text": "", "logprob": -2.203125, "special": false },
         { "id": 96, "text": "أ", "logprob": -1.2207031, "special": false },
         { "id": 45672, "text": "ك", "logprob": -0.40942383, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.5161133, "special": false },
         { "id": 15065, "text": "د", "logprob": -0.092163086, "special": false },
         { "id": 17535, "text": "", "logprob": -1.4130859, "special": false },
         { "id": 96, "text": " أ", "logprob": -1.4550781, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.6098633, "special": false },
         { "id": 207, "text": " ", "logprob": -1.1396484, "special": false },
         { "id": 20000, "text": "ي", "logprob": -1.6914062, "special": false },
         { "id": 146, "text": "", "logprob": -1.9453125, "special": false },
         { "id": 114, "text": "ض", "logprob": -1.4912109, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.7949219, "special": false },
         { "id": 17535, "text": "", "logprob": -1.2480469, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.8979492, "special": false },
         { "id": 16213, "text": "ن", "logprob": -1.0400391, "special": false },
         { "id": 207, "text": " ", "logprob": -0.68896484, "special": false },
         { "id": 20000, "text": "ي", "logprob": -1.0244141, "special": false },
         { "id": 146, "text": "", "logprob": -1.0244141, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.30981445, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.1809082, "special": false },
         { "id": 17535, "text": "", "logprob": -0.72753906, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.27734375, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.2861328, "special": false },
         { "id": 207, "text": " ", "logprob": -0.22973633, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.13024902, "special": false },
         { "id": 146, "text": "", "logprob": -0.038269043, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.0049705505, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00299263, "special": false },
         { "id": 17535, "text": "", "logprob": -0.16967773, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.024429321, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.055999756, "special": false },
         { "id": 207, "text": " ", "logprob": -0.15710449, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.03768921, "special": false },
         { "id": 146, "text": "", "logprob": -0.011924744, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.0014295578, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.0009217262, "special": false },
         { "id": 17535, "text": "", "logprob": -0.1282959, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.012008667, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.030960083, "special": false },
         { "id": 207, "text": " ", "logprob": -0.13574219, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.024093628, "special": false },
         { "id": 146, "text": "", "logprob": -0.0073509216, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.00074386597, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.000497818, "special": false },
         { "id": 17535, "text": "", "logprob": -0.10437012, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0069885254, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.021011353, "special": false },
         { "id": 207, "text": " ", "logprob": -0.11779785, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.018356323, "special": false },
         { "id": 146, "text": "", "logprob": -0.0053367615, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.0004632473, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.0003387928, "special": false },
         { "id": 17535, "text": "", "logprob": -0.08605957, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0047798157, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.016479492, "special": false },
         { "id": 207, "text": " ", "logprob": -0.10668945, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.015075684, "special": false },
         { "id": 146, "text": "", "logprob": -0.0041236877, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.0003376007, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00027298927, "special": false },
         { "id": 17535, "text": "", "logprob": -0.0725708, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0033473969, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.012710571, "special": false },
         { "id": 207, "text": " ", "logprob": -0.095581055, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.013015747, "special": false },
         { "id": 146, "text": "", "logprob": -0.0034427643, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.00025820732, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00022995472, "special": false },
         { "id": 17535, "text": "", "logprob": -0.062042236, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0025424957, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.009666443, "special": false },
         { "id": 207, "text": " ", "logprob": -0.08691406, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.011428833, "special": false },
         { "id": 146, "text": "", "logprob": -0.0028762817, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.00019574165, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.0001950264, "special": false },
         { "id": 17535, "text": "", "logprob": -0.05606079, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0019874573, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.008216858, "special": false },
         { "id": 207, "text": " ", "logprob": -0.08013916, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.010299683, "special": false },
         { "id": 146, "text": "", "logprob": -0.0025196075, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.00015699863, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00017774105, "special": false },
         { "id": 17535, "text": "", "logprob": -0.050445557, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0017375946, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.006965637, "special": false },
         { "id": 207, "text": " ", "logprob": -0.07550049, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.009254456, "special": false },
         { "id": 146, "text": "", "logprob": -0.0022563934, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.00013005733, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.0001603365, "special": false },
         { "id": 17535, "text": "", "logprob": -0.044555664, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0014829636, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.0060157776, "special": false },
         { "id": 207, "text": " ", "logprob": -0.06945801, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.008468628, "special": false },
         { "id": 146, "text": "", "logprob": -0.0020771027, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.00010788441, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00015366077, "special": false },
         { "id": 17535, "text": "", "logprob": -0.04071045, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0012512207, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.0052604675, "special": false },
         { "id": 207, "text": " ", "logprob": -0.06347656, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.007663727, "special": false },
         { "id": 146, "text": "", "logprob": -0.001950264, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.00009858608, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.0001436472, "special": false },
         { "id": 17535, "text": "", "logprob": -0.038330078, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0010757446, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.004524231, "special": false },
         { "id": 207, "text": " ", "logprob": -0.059906006, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.0071411133, "special": false },
         { "id": 146, "text": "", "logprob": -0.0018062592, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.00008201599, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00013720989, "special": false },
         { "id": 17535, "text": "", "logprob": -0.035003662, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0010290146, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.004020691, "special": false },
         { "id": 207, "text": " ", "logprob": -0.055908203, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.006664276, "special": false },
         { "id": 146, "text": "", "logprob": -0.0017576218, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.000073194504, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00013172626, "special": false },
         { "id": 17535, "text": "", "logprob": -0.032287598, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.000910759, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.0036144257, "special": false },
         { "id": 207, "text": " ", "logprob": -0.052581787, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.006275177, "special": false },
         { "id": 146, "text": "", "logprob": -0.0016613007, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.00006711483, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00012528896, "special": false },
         { "id": 17535, "text": "", "logprob": -0.030029297, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0008749962, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.0033073425, "special": false },
         { "id": 207, "text": " ", "logprob": -0.049957275, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.0059165955, "special": false },
         { "id": 146, "text": "", "logprob": -0.001581192, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.000061154366, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00012040138, "special": false },
         { "id": 17535, "text": "", "logprob": -0.028427124, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0008134842, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.0029716492, "special": false },
         { "id": 207, "text": " ", "logprob": -0.048065186, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.0055007935, "special": false },
         { "id": 146, "text": "", "logprob": -0.0016021729, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.00005710125, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00011897087, "special": false },
         { "id": 17535, "text": "", "logprob": -0.027435303, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0007419586, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.002937317, "special": false },
         { "id": 207, "text": " ", "logprob": -0.045288086, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.005268097, "special": false },
         { "id": 146, "text": "", "logprob": -0.0015268326, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.000049471855, "special": false },
         {
         "id": 26485,
         "text": "ع",
         "logprob": -0.000116467476,
         "special": false
         },
         { "id": 17535, "text": "", "logprob": -0.0259552, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.000688076, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.002626419, "special": false },
         { "id": 207, "text": " ", "logprob": -0.042816162, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.0049362183, "special": false },
         { "id": 146, "text": "", "logprob": -0.0014801025, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.00004887581, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00011277199, "special": false },
         { "id": 17535, "text": "", "logprob": -0.024841309, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.000682354, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.0025501251, "special": false },
         { "id": 207, "text": " ", "logprob": -0.042144775, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.0047416687, "special": false },
         { "id": 146, "text": "", "logprob": -0.0014600754, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.000046014786, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00010919571, "special": false },
         { "id": 17535, "text": "", "logprob": -0.023483276, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0006299019, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.0023880005, "special": false },
         { "id": 207, "text": " ", "logprob": -0.039611816, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.00466156, "special": false },
         { "id": 146, "text": "", "logprob": -0.0013542175, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.000041484833, "special": false },
         {
         "id": 26485,
         "text"arabic: "ع",
         "logprob": -0.000106334686,
         "special": false
         },
         { "id": 17535, "text": "", "logprob": -0.023269653, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0006098747, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.002243042, "special": false },
         { "id": 207, "text": " ", "logprob": -0.038360596, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.0044136047, "special": false },
         { "id": 146, "text": "", "logprob": -0.0013628006, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.000039815903, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00010180473, "special": false },
         { "id": 17535, "text": "", "logprob": -0.022399902, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.00058984756, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.0022029877, "special": false },
         { "id": 207, "text": " ", "logprob": -0.03744507, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.00415802, "special": false },
         { "id": 146, "text": "", "logprob": -0.0013427734, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.0000371933, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00010359287, "special": false },
         { "id": 17535, "text": "", "logprob": -0.021057129, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0005173683, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.0019779205, "special": false },
         { "id": 207, "text": " ", "logprob": -0.035980225, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.0039520264, "special": false },
         { "id": 146, "text": "", "logprob": -0.0012645721, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.000034689903, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00010192394, "special": false },
         { "id": 17535, "text": "", "logprob": -0.02053833, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.00050497055, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.0019207001, "special": false },
         { "id": 207, "text": " ", "logprob": -0.034851074, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.0038108826, "special": false },
         { "id": 146, "text": "", "logprob": -0.0012216568, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.000030875206, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00009846687, "special": false },
         { "id": 17535, "text": "", "logprob": -0.02029419, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0005040169, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.0018539429, "special": false },
         { "id": 207, "text": " ", "logprob": -0.03326416, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.0036506653, "special": false },
         { "id": 146, "text": "", "logprob": -0.0011854172, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.000030636787, "special": false },
         {
         "id": 26485,
         "text": "ع",
         "logprob": -0.000097990036,
         "special": false
         },
         { "id": 17535, "text": "", "logprob": -0.018920898, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.00046873093, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.0018253326, "special": false },
         { "id": 207, "text": " ", "logprob": -0.033294678, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.0035190582, "special": false },
         { "id": 146, "text": "", "logprob": -0.0011663437, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.000027775764, "special": false },
         {
         "id": 26485,
         "text": "ع",
         "logprob": -0.000094771385,
         "special": false
         },
         { "id": 17535, "text": "", "logprob": -0.018554688, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.000436306, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.0016956329, "special": false },
         { "id": 207, "text": " ", "logprob": -0.03225708, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.0034160614, "special": false },
         { "id": 146, "text": "", "logprob": -0.0011491776, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.000027537346, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00009381771, "special": false },
         { "id": 17535, "text": "", "logprob": -0.018371582, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.00046014786, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.0016422272, "special": false },
         { "id": 207, "text": " ", "logprob": -0.031082153, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.0033130646, "special": false },
         { "id": 146, "text": "", "logprob": -0.0011634827, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.000025629997, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00009512901, "special": false },
         { "id": 17535, "text": "", "logprob": -0.017745972, "special": false },
         { "id": 96, "text": " أ", "logprob": -0.0004081726, "special": false },
         { "id": 16213, "text": "ن", "logprob": -0.0015878677, "special": false },
         { "id": 207, "text": " ", "logprob": -0.029693604, "special": false },
         { "id": 20000, "text": "ي", "logprob": -0.00315094, "special": false },
         { "id": 146, "text": "", "logprob": -0.001080513, "special": false },
         { "id": 114, "text": "ض", "logprob": -0.00002360344, "special": false },
         { "id": 26485, "text": "ع", "logprob": -0.00009047985, "special": false },
         { "id": 17535, "text": "", "logprob": -0.017074585, "special": false }
      ]
   }
   }
   ```

   </details>

Setting `max_new_tokens` to 255 or 257 fixes it for this specific prompt

### Expected behavior

I don't know what the expected behaviour for the last character should be (I'd assume just cut leave it out), but the start of the string should at least remain unchanged by one rogue byte at the end.

**Comment by randxie** [comment]
The issue comes from a combination of [this line](https://github.com/huggingface/text-generation-inference/blob/main/server/text_generation_server/models/model.py#L87):
`not new_text.endswith("�")`

And [these lines](https://github.com/huggingface/text-generation-inference/blob/ccd5725a0c0b2ef151d317c86d1f52ad038bbae9/server/text_generation_server/models/causal_lm.py#L642-L652). 

TLDR: when your stop criterion is hit, the final token ends at "�", that indicates your last token is not a valid utf-8 char. And TGI just return empty string in the `generated_text` field, even the tokens have been generated.

**Comment by Vinno97** [comment]
This is still an active bug that can affect real-world deployments
